src/goto-symex/symex_target_equation.cpp

```diff
@@ -351,11 +351,11 @@ void symex_target_equationt::convert(
 }
 
 void symex_target_equationt::convert_assignments(
-  decision_proceduret &decision_procedure) const
+  decision_proceduret &decision_procedure)
 {
-  for(const auto &step : SSA_steps)
+  for(auto &step : SSA_steps)
   {
-    if(step.is_assignment() && !step.ignore)
+    if(step.is_assignment() && !step.ignore && !step.converted)
```

Comment:
Instead of adding this info to every step, would it be more efficient to store a conversion frontier as a single iterator member? The current implementation increases the storage size of each `SSA_stept` and requires repeated traversal of steps have have been seen before. With an `optionalt<SSA_stepst::const_iterator>` we would just store a pointer to the last converted step.

Reply:
There is no conversion frontier due to slicing.

Reply:
Ah, of course.